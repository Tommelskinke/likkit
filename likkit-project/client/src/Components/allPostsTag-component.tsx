import * as React from 'react';
import { Component } from 'react-simplified';
import { Card, Row, Column, Form, Button, NavBar, upLikk, downLikk } from '../widgets';
import taskService, { Question, Favorites } from '../question-service';
import { createHashHistory } from 'history';
import PrettyPreview from './prettyPreview-component';

const history = createHashHistory();

export class AllPostsTag extends Component {
  search: string = '';
  user_id: number = Number(sessionStorage.getItem('user_id'));
  posts: Question[] = [];
  userFavorites: Favorites[] = [];
  selectedOption: string = 'popular';

  handleSortChange = (event: any) => {
    const selectedOption = event.target.value;
    this.selectedOption = selectedOption;

    if (selectedOption === '1') {
      taskService.questionGetAllTag(1).then((posts) => (this.posts = posts));
    } else if (selectedOption === '2') {
      taskService.questionGetAllTag(2).then((posts) => (this.posts = posts));
    } else if (selectedOption === '3') {
      taskService.questionGetAllTag(3).then((posts) => (this.posts = posts));
    } else if (selectedOption === '4') {
      taskService.questionGetAllTag(4).then((posts) => (this.posts = posts));
    }
    this.forceUpdate(); // Trigger a re-render
  };

  handleUpvote = (questionId: number) => {
    taskService
      .upvoteQuestion(questionId)
      .then(() => {
        this.handleSortChange({ target: { value: this.selectedOption } });
      })
      .catch((error) => {
        console.error('Error upvoting question:', error);
      });
  };

  handleDownvote = (questionId: number) => {
    taskService
      .downvoteQuestion(questionId)
      .then(() => {
        this.handleSortChange({ target: { value: this.selectedOption } });
      })
      .catch((error) => {
        console.error('Error downvoting question:', error);
      });
  };

  render() {
    return (
      <div
        className="background"
        style={{
          backgroundImage: 'linear-gradient(180deg, rgb(110, 160, 175), rgb(15, 40, 60))',
          display: 'flex',
          flexDirection: 'column',
          alignItems: 'stretch',
        }}
      >
        <div
          style={{
            margin: '2%',
          }}
        >
          <Card title="" width="100%" backgroundColor="rgb(90,90,90)">
            <Row marginBottom={5}>
              <div
                style={{ color: 'white', fontWeight: 'bold', fontSize: '20px', marginLeft: '5px' }}
              >
                <Column>Sort by tags:</Column>
                <Column>
                  <div style={{ marginLeft: '10px' }}>
                    <select
                      id="dropdown"
                      style={{
                        appearance: 'none',
                        background: '#5A5A5A',
                        color: '#ffffff',
                        cursor: 'pointer',
                        border: '1px solid #ffffff',
                        borderRadius: '10px',
                        padding: '5px',
                      }}
                      onChange={this.handleSortChange}
                      value={this.selectedOption}
                    >
                      <option value="1">Solved</option>
                      <option value="2">JavaScript</option>
                      <option value="3">Rust</option>
                      <option value="4">Linux</option>
                    </select>
                  </div>
                </Column>
                <Column right>
                  <div style={{ marginRight: '5px' }}>Posts with this tag: {this.posts.length}</div>
                </Column>
              </div>
            </Row>

            <Card title="" width="100%" backgroundColor="rgb(70,70,70)" marginBottom={-3}>
              <Row>
                {this.posts.map((post, i) => (
                  <Card
                    title=""
                    width="100%"
                    backgroundColor="rgb(60,60,60)"
                    marginBottom={3}
                    key={i}
                  >
                    <Row>
                      <Column width={2}>
                        <div
                          style={{
                            color: 'white',
                            fontWeight: 'bold',
                            fontSize: '25px',
                            display: 'flex',
                            flexDirection: 'row',
                            alignItems: 'center',
                            alignContent: 'center',
                          }}
                        >
                          <Column width={4}>
                            <Button.Vote onClick={() => this.handleUpvote(post.question_id)}>
                              {upLikk}
                            </Button.Vote>
                          </Column>
                          <Column width={4}>
                            <p style={{ margin: '0 10px' }}>{post.upvotes - post.downvotes}</p>
                          </Column>
                          <Column width={4}>
                            <Button.Vote onClick={() => this.handleDownvote(post.question_id)}>
                              {downLikk}
                            </Button.Vote>
                          </Column>
                        </div>
                      </Column>
                      <Column width={8} none>
                        {/*Koden her gir error siden den ikke liker at vi har en knapp og annet innhold inne i en knapp*/}
                        <div
                          style={{
                            display: 'flex',
                            flexDirection: 'column',
                            justifyContent: 'center',
                            alignItems: 'center',
                            textAlign: 'center',
                            cursor: 'pointer',
                          }}
                          onClick={() => history.push('/posts/' + post.question_id)}
                        >
                          <div style={{ color: 'white', fontWeight: 'bold', fontSize: '25px' }}>
                            <p style={{ alignItems: 'center', alignContent: 'center' }}>
                              {post.title}
                            </p>
                            <div style={{ fontSize: '14px', fontWeight: 'normal' }}>
                              <PrettyPreview htmlContent={post.content} maxLength={100} />
                            </div>
                          </div>
                        </div>
                      </Column>
                      <Column width={2} right>
                        <div>
                          {this.userFavorites.some(
                            (favorite) =>
                              favorite.question_id == post.question_id &&
                              favorite.answer_id == null,
                          ) ? (
                            <img
                              style={{ cursor: 'pointer' }}
                              src="https://upload.wikimedia.org/wikipedia/commons/thumb/2/29/Gold_Star.svg/1024px-Gold_Star.svg.png"
                              alt="Filled picture of gold star, indicates favorites"
                              onClick={() => {
                                taskService
                                  .removeFavorite(this.user_id, post.question_id, null)
                                  .then(() => {
                                    window.location.reload();
                                  });
                              }}
                            />
                          ) : (
                            <img
                              style={{ cursor: 'pointer' }}
                              src="https://upload.wikimedia.org/wikipedia/commons/thumb/1/18/Five-pointed_star.svg/800px-Five-pointed_star.svg.png"
                              alt="Empty picture of star, indicates not favorites"
                              onClick={() => {
                                taskService
                                  .addFavorite(this.user_id, post.question_id, null)
                                  .then(() => {
                                    window.location.reload();
                                  });
                              }}
                            />
                          )}
                        </div>
                      </Column>
                    </Row>
                  </Card>
                ))}
              </Row>
            </Card>
          </Card>
        </div>
      </div>
    );
  }
  mounted() {
    taskService.questionGetAllTag(1).then((posts) => (this.posts = posts));
    taskService.getUserFavorites(this.user_id).then((userFavorites) => {
      this.userFavorites = userFavorites;
    });
  }
}
