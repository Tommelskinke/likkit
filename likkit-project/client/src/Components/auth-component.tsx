import * as React from 'react';
import { Component } from 'react-simplified';
import { Card, Row, Column, Form, Button } from '../widgets';

export class Login extends Component<{ match: { params: { id: number } } }> {
  username = '';
  password = '';
  GoogleLoginButton = () => {
    return (
      <a style={{ color: 'white' }} href="http://localhost:3000/oauth2/google">
        Log in as User
      </a>
    );
  };
  render() {
    return (
      <div
        style={{
          backgroundImage: 'linear-gradient(180deg, rgb(110, 160, 175), rgb(15, 40, 60))',
          display: 'flex',
          flexDirection: 'column',
          alignItems: 'center',
          justifyContent: 'center',
          height: '100vh',
        }}
      >
        <div
          style={{
            margin: '1%',
            color: 'white',
            fontWeight: 'bold',
            fontSize: '20px',
          }}
        >
          <Card title="Log in as Admin" width="100%" backgroundColor="rgb(90,90,90)">
            <Row>
              <Column width={9}>
                <Form.Input
                  type="text"
                  value={this.username}
                  onChange={(event) => (this.username = event.currentTarget.value)}
                  placeholder="username"
                />
                <Form.Input
                  type="password"
                  value={this.password}
                  onChange={(event) => (this.password = event.currentTarget.value)}
                  placeholder="password"
                />
                <Button.Success
                  onClick={() => {
                    console.log(this.username);
                  }}
                >
                  Login
                </Button.Success>
              </Column>
              <Column width={3}>
                <this.GoogleLoginButton />
              </Column>
            </Row>
          </Card>
        </div>
      </div>
    );
  }
}
