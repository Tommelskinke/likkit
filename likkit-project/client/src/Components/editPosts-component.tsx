import * as React from 'react';
import { Component } from 'react-simplified';
import taskService, { Question, Tag } from '../question-service';
import { Card, Row, Column, Form, Button } from '../widgets';
import { createHashHistory } from 'history';
import EditorComponent from './editor-component';

const history = createHashHistory();

export class EditPost extends Component<{ match: { params: { id: number } } }> {
  title: string = '';
  content: string = '';
  tags: Tag[] = [];
  user_id: number = Number(sessionStorage.getItem('user_id'));

  questionNew: Question = {
    username: '',
    question_id: 0,
    user_id: 0,
    title: '',
    content: '',
    created_at: '',
    upvotes: 0,
    downvotes: 0,
  };

  questionSolved: boolean = false;
  questionJS: boolean = false;
  questionRust: boolean = false;
  questionLinux: boolean = false;

  render() {
    return (
      <div className="background">
        <div
          style={{
            margin: '1%',
            marginLeft: '15%',
            marginRight: '15%',
          }}
        >
          <Card title="" width="100%" backgroundColor="rgb(90,90,90)">
            <Row marginBottom={3}>
              <Column width={2}>
                <div style={{ color: 'white', fontWeight: 'bold', fontSize: '20px' }}>Title </div>
              </Column>
              <Column width={7} none>
                <Form.Input
                  type="text"
                  value={this.questionNew.title}
                  onChange={(event) => (this.questionNew.title = event.currentTarget.value)}
                  placeholder="Title..."
                />
              </Column>
            </Row>
            <Row marginBottom={3}>
              <Column width={1}>
                <div style={{ color: 'white', fontSize: '14px' }}>Solved?</div>
              </Column>
              <Column width={2}>
                <Form.Checkbox
                  checked={this.questionSolved}
                  onChange={(event) =>
                    (this.questionSolved = event.currentTarget.checked ? true : false)
                  }
                />
              </Column>
              <Column width={1}>
                <div style={{ color: 'white', fontSize: '14px' }}>JavaScript</div>
              </Column>
              <Column width={2}>
                <Form.Checkbox
                  checked={this.questionJS}
                  onChange={(event) =>
                    (this.questionJS = event.currentTarget.checked ? true : false)
                  }
                />
              </Column>
              <Column width={1}>
                <div style={{ color: 'white', fontSize: '14px' }}>Rust</div>
              </Column>
              <Column width={2}>
                <Form.Checkbox
                  checked={this.questionRust}
                  onChange={(event) =>
                    (this.questionRust = event.currentTarget.checked ? true : false)
                  }
                />
              </Column>
              <Column width={1}>
                <div style={{ color: 'white', fontSize: '14px' }}>Linux</div>
              </Column>
              <Column width={2}>
                <Form.Checkbox
                  checked={this.questionLinux}
                  onChange={(event) =>
                    (this.questionLinux = event.currentTarget.checked ? true : false)
                  }
                />
              </Column>
            </Row>
            <Row marginBottom={5}>
              <Column width={2}></Column>
              <Column width={9} none>
                <EditorComponent
                  onContentChange={(content: string) => {
                    this.questionNew.content = content;
                  }}
                />
              </Column>
            </Row>
            <Row marginBottom={2}>
              <Column right>
                <Button.Success
                  onClick={() => {
                    if (this.title.length <= 255) {
                      taskService
                        .questionEdit(
                          this.questionNew.title,
                          this.questionNew.content,
                          this.questionNew.question_id,
                        )
                        .then(() => {});

                      //gjÃ¸r dette om til en forEach loop
                      if (this.questionSolved) {
                        taskService.questionTagCreate(this.questionNew.question_id, 1);
                      } else {
                        taskService.questionTagRemove(this.questionNew.question_id, 1);
                      }
                      if (this.questionJS) {
                        taskService.questionTagCreate(this.questionNew.question_id, 2);
                      } else {
                        taskService.questionTagRemove(this.questionNew.question_id, 2);
                      }
                      if (this.questionRust) {
                        taskService.questionTagCreate(this.questionNew.question_id, 3);
                      } else {
                        taskService.questionTagRemove(this.questionNew.question_id, 3);
                      }
                      if (this.questionLinux) {
                        taskService.questionTagCreate(this.questionNew.question_id, 4);
                      } else {
                        taskService.questionTagRemove(this.questionNew.question_id, 4);
                      }
                      taskService.questionGetNewest().then(() => {});
                      setTimeout(() => {
                        history.push('/posts/' + this.questionNew.question_id);
                      }, 100);
                    } else {
                      alert('The title cant be more than 255 characters!');
                    }
                  }}
                >
                  Post
                </Button.Success>
              </Column>
            </Row>
          </Card>
        </div>
      </div>
    );
  }
  mounted() {
    taskService
      .questionGet(this.props.match.params.id)
      .then((question) => (this.questionNew = question));
    taskService
      .questionTagGet(this.props.match.params.id)
      .then((tags) => (this.tags = tags))
      .then(() => {
        this.tags.forEach((tag) => {
          switch (tag.tag_id) {
            case 1:
              this.questionSolved = true;
              break;
            case 2:
              this.questionJS = true;
              break;
            case 3:
              this.questionRust = true;
              break;
            case 4:
              this.questionLinux = true;
              break;
            // Add more cases if needed for additional tags
            default:
              break;
          }
        });
      });
  }
}
