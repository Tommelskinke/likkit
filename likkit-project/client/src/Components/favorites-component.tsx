import * as React from 'react';
import { Component } from 'react-simplified';
import { Alert, Card, Row, Column, Form, Button, NavBar } from '../widgets';

export class FavoriteList extends Component {
  //posts: string[] = ['Post1', 'post2']; //Denne må endres te rett type seinere
  search: string = '';

  posts = [
    {
      text: 'Se på kjeksene mine!!!!',
      imageLink: 'https://i.scdn.co/image/ab6761610000e5eba3a7cba23d68a4e73c3b8155',
      likks: '30',
    },
    {
      text: 'anyways... COYS',
      imageLink: 'https://i.ytimg.com/vi/H8ZnIAama1A/maxresdefault.jpg',
      likks: '15',
    },
    {
      text: 'Fo.',
      likks: '45',
    },
  ];

  alerttest = () => {
    alert('SHREK');
  };

  render() {
    return (
      <div
        style={{
          backgroundImage: 'linear-gradient(180deg, rgb(110, 160, 175), rgb(15, 40, 60))',
          display: 'flex',
          flexDirection: 'column',
          alignItems: 'stretch',
          height: '100vh', //bør få denne til å scale
        }}
      >
        <div
          style={{
            margin: '2%',
          }}
        >
          <Card title="" width="100%" backgroundColor="rgb(90,90,90)">
            <Row marginBottom={5}>
              <div
                style={{ color: 'white', fontWeight: 'bold', fontSize: '20px', marginLeft: '5px' }}
              >
                <Column>Sort by:</Column>
                <Column>
                  <div style={{ marginLeft: '10px' }}>
                    <select
                      id="dropdown"
                      style={{
                        appearance: 'none',
                        background: '#5A5A5A',
                        color: '#ffffff',
                        cursor: 'pointer',
                        border: '1px solid #ffffff',
                        borderRadius: '10px',
                        padding: '5px',
                      }}
                    >
                      <option value="">Popular</option>
                      <option value="op1">Best</option>
                    </select>
                  </div>
                </Column>
                <Column right>
                  <div
                    style={{
                      appearance: 'none',
                      background: '#5A5A5A',
                      color: '#ffffff',
                      cursor: 'pointer',
                      border: '1px solid #ffffff',
                      borderRadius: '10px',
                      padding: '5px',
                    }}
                  >
                    <select
                      id="filterdropdown"
                      style={{
                        appearance: 'none',
                        border: '#5A5A5A',
                        background: '#5A5A5A',
                        color: '#ffffff',
                        cursor: 'pointer',
                      }}
                    >
                      <option value="none">None</option>
                      <option value="tagBesvart">Besvarte spørsmål</option>
                      <option value="tagUbesvart">Ubesvarte spørsmål</option>
                      <option value="tagMine">Mine spørsmål</option>
                      <option value="tagJS">Javascript</option>
                      <option value="tagC">C</option>
                    </select>
                  </div>
                </Column>
                <Column right>
                  <div style={{ marginRight: '5px' }}>Filters:</div>
                </Column>
              </div>
            </Row>

            <Card title="" width="100%" backgroundColor="rgb(70,70,70)" marginBottom={-3}>
              <Row>
                <Column>
                  {this.posts.map((post, i) => (
                    <Card
                      title=""
                      width="100%"
                      backgroundColor="rgb(60,60,60)"
                      marginBottom={3}
                      key={i}
                    >
                      <Button.Post onClick={this.alerttest}>
                        <div style={{ color: 'white', fontWeight: 'bold', fontSize: '25px' }}>
                          <p style={{ position: 'absolute', top: '0', left: '10' }}>{post.likks}</p>
                          <p>{post.text}</p>
                        </div>
                      </Button.Post>
                    </Card>
                  ))}
                </Column>
                <Column right>
                  <Form.Input
                    type="text"
                    value={this.search}
                    onChange={(event) => (this.search = event.currentTarget.value)}
                    placeholder="Search..."
                  />
                </Column>
              </Row>
            </Card>
          </Card>
        </div>
      </div>
    );
  }
}
