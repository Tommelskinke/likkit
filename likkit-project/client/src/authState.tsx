import React from 'react';

interface UserData {
  user_id: number;
  email?: string;
  user_type: string;
  username?: string;
  user_pfp?: string;
}

// Example using Fetch API in a frontend JavaScript
fetch('/api/userinfo')
  .then((response) => {
    if (!response.ok) {
      throw new Error('Failed to fetch user info');
    }
    return response.json();
  })
  .then((userData) => {
    console.log(userData);
    sessionStorage.setItem('user_id', userData.user_id);
    sessionStorage.setItem('email', userData.email);
    sessionStorage.setItem('user_type', userData.user_type);
    sessionStorage.setItem('username', userData.username);
    sessionStorage.setItem('user_pfp', userData.user_pfp);
    // Now you have user data, including the profile picture URL
    // You can display this data in your UI
  })
  .catch((error) => console.error(error));

export function handleLogout() {
  fetch('/logout')
    .then((response) => {
      if (response.ok) {
        window.location.href = '/';
      } else {
        console.error('Failed to log out');
      }
    })
    .catch((error) => {
      console.error('Logout error: ', error);
      // Handle error case
    });
}

// Create a context with an initial empty state
export const UserContext = React.createContext<UserData | null>(null);

// Export the provider as a separate component for convenience
export const UserProvider = UserContext.Provider;
