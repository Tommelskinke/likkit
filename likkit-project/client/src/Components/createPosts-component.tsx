import * as React from 'react';
import { UserContext, UserProvider } from '../authState';
import { Component } from 'react-simplified';
import taskService, { Question } from '../question-service';
import { Alert, Card, Row, Column, Form, Button, NavBar } from '../widgets';
import { createHashHistory } from 'history';

const history = createHashHistory();

export class CreatePost extends Component {
  title: string = '';
  content: string = '';
  tags: string = '';
  user_id: number = Number(sessionStorage.getItem('user_id'));

  questionNew: Question = {
    username: '',
    question_id: 0,
    user_id: 0,
    title: "",
    content: "",
    created_at: "",
    upvotes: 0,
    downvotes: 0,
    karma: 0
  };
  
  questionSolved: boolean = false;
  questionJS: boolean = false;
  questionRust: boolean = false;
  questionLinux: boolean = false;

  render() {
    return (
          <div
            style={{
              backgroundImage: 'linear-gradient(180deg, rgb(110, 160, 175), rgb(15, 40, 60))',
              display: 'flex',
              flexDirection: 'column',
              alignItems: 'stretch',
              height: '150vh',
            }}
          >
            <div
              style={{
                margin: '1%',
                marginLeft: '15%',
                marginRight: '15%',
              }}
            >
              <Card title="" width="100%" backgroundColor="rgb(90,90,90)">
                <Row marginBottom={3}>
                  <Column width={2}>
                    <div style={{ color: 'white', fontWeight: 'bold', fontSize: '20px' }}>Title </div>
                  </Column>
                  <Column width={7} none>
                    <Form.Input
                      type="text"
                      value={this.title}
                      onChange={(event) => (this.title = event.currentTarget.value)}
                      placeholder="Title..."
                    />
                  </Column>
                </Row>
                <Row marginBottom={3}>
                  <Column width={1}>
                    <div style={{ color: 'white', fontSize: '14px' }}>Solved?</div>
                  </Column>
                  <Column width={2}>
                    <Form.Checkbox
                      checked={this.questionSolved}
                      onChange={(event) =>
                        (this.questionSolved = event.currentTarget.checked ? true : false)
                      }
                    />
                  </Column>
                  <Column width={1}>
                    <div style={{ color: 'white', fontSize: '14px' }}>JavaScript</div>
                  </Column>
                  <Column width={2}>
                    <Form.Checkbox
                      checked={this.questionJS}
                      onChange={(event) =>
                        (this.questionJS = event.currentTarget.checked ? true : false)
                      }
                    />
                  </Column>
                  <Column width={1}>
                    <div style={{ color: 'white', fontSize: '14px' }}>Rust</div>
                  </Column>
                  <Column width={2}>
                    <Form.Checkbox
                      checked={this.questionRust}
                      onChange={(event) =>
                        (this.questionRust = event.currentTarget.checked ? true : false)
                      }
                    />
                  </Column>
                  <Column width={1}>
                    <div style={{ color: 'white', fontSize: '14px' }}>Linux</div>
                  </Column>
                  <Column width={2}>
                    <Form.Checkbox
                      checked={this.questionLinux}
                      onChange={(event) =>
                        (this.questionLinux = event.currentTarget.checked ? true : false)
                      }
                    />
                  </Column>
                </Row>
                <Row marginBottom={5}>
                  <Column width={2}></Column>
                  <Column width={9} none>
                    <Form.Textarea
                      type="text"
                      value={this.content}
                      onChange={(event) => (this.content = event.currentTarget.value)}
                      placeholder="Write your post here..."
                      style={{ height: '55vh' }}
                    />
                  </Column>
                </Row>
                <Row marginBottom={2}>
                  <Column right>
                    <Button.Success
                      onClick={() => {

                        if ((this.title.length <= 255) && (this.content.length <= 255)) {
                          taskService.questionCreate(this.user_id, this.title, this.content).then(() => { 
                          });
                          
                          //gjÃ¸r dette om til en forEach loop
                          if (this.questionSolved) {
                            taskService.questionTagCreate((this.questionNew.question_id + 1), 1)
                          }
                          if (this.questionJS) {
                            taskService.questionTagCreate((this.questionNew.question_id + 1), 2)
                          }
                          if (this.questionRust) {
                            taskService.questionTagCreate((this.questionNew.question_id + 1), 3)
                          }
                          if (this.questionLinux) {
                            taskService.questionTagCreate((this.questionNew.question_id + 1), 4)
                          }
                          taskService.questionGetNewest().then(() => {
                          
                          })
                          setTimeout(() => {  history.push('/posts/' + (this.questionNew.question_id + 1)) }, 100);
                                   
                        } else {
                          alert("The title and/or content cant be more than 255 characters!")
                        }
                      }}
                    >
                      Post
                    </Button.Success>
                  </Column>
                </Row>
              </Card>
            </div>
          </div>
    );
  }
  mounted() {
    taskService.questionGetNewest().then((questionNew) => (this.questionNew = questionNew));
  }
}
