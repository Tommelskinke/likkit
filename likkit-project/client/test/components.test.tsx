import * as React from 'react';
import { Menu, Home } from '../src/Components/homepage-component';
import { FavoriteList } from '../src/Components/favorites-component';
import { shallow } from 'enzyme';
import { EditPost } from '../src/Components/editPosts-component';
import { CreatePost } from '../src//Components/createPosts-component';
import { AllPostsTag } from '../src/Components/allPostsTag-component';
import { Form, Button, Column, Alert } from '../src/widgets';
import { ViewPost } from '../src/Components/viewPosts-component';
import { Login } from '../src//Components/auth-component';
import { NavLink } from 'react-router-dom';
import { UserProvider } from '../src/authState';
import { AllPosts } from '../src/Components/allPosts-component';
import PrettyPreview from '../src/Components/prettyPreview-component';

describe('Component tests', () => {
  test('FavoriteList', (done) => {
    const wrapper = shallow(<FavoriteList />);

    // Wait for events to complete
    setTimeout(() => {
      expect(wrapper).toMatchSnapshot();
      done();
    });
  });
  test('Login', (done) => {
    const match = { params: { id: 1 } }; // replace 123 with an appropriate id
    const wrapper = shallow(<Login match={match} />);

    // Wait for events to complete
    setTimeout(() => {
      expect(wrapper).toMatchSnapshot();
      done();
    });
  });
  test('All Posts', (done) => {
    const wrapper = shallow(<AllPosts />);

    // Wait for events to complete
    setTimeout(() => {
      expect(wrapper).toMatchSnapshot();
      done();
    });
  });
  test('All Post tags', (done) => {
    const wrapper = shallow(<AllPostsTag />);

    // Wait for events to complete
    setTimeout(() => {
      expect(wrapper).toMatchSnapshot();
      done();
    });
  });
});

describe('PrettyPreview', () => {
  it('should strip HTML tags', () => {
    const wrapper = shallow(<PrettyPreview htmlContent="<p>Hello, world!</p>" />);
    expect(wrapper.text()).toEqual('Hello, world!');
  });

  it('should truncate text', () => {
    const wrapper = shallow(<PrettyPreview htmlContent="<p>Hello, world!</p>" maxLength={5} />);
    expect(wrapper.text()).toEqual('Hello...');
  });

  it('should render correctly', () => {
    const wrapper = shallow(<PrettyPreview htmlContent="<p>Hello, world!</p>" />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('widget tests', () => {
  test('alert', (done) => {
    const wrapper = shallow(<Alert />);

    // Wait for events to complete
    setTimeout(() => {
      expect(wrapper).toMatchSnapshot();
      done();
    });
  });
  test('Logiasffsan', (done) => {
    const match = { params: { id: 1 } }; // replace 123 with an appropriate id
    const wrapper = shallow(<ViewPost match={match} />);

    // Wait for events to complete
    setTimeout(() => {
      expect(wrapper).toMatchSnapshot();
      done();
    });
  });
});
