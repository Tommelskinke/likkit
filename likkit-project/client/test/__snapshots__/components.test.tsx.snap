// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component tests All Post tags 1`] = `
<div
  className="background"
  style={
    {
      "alignItems": "stretch",
      "backgroundImage": "linear-gradient(180deg, rgb(110, 160, 175), rgb(15, 40, 60))",
      "display": "flex",
      "flexDirection": "column",
    }
  }
>
  <div
    style={
      {
        "margin": "2%",
      }
    }
  >
    <Card
      backgroundColor="rgb(90,90,90)"
      title=""
      width="100%"
    >
      <Row
        marginBottom={5}
      >
        <div
          style={
            {
              "color": "white",
              "fontSize": "20px",
              "fontWeight": "bold",
              "marginLeft": "5px",
            }
          }
        >
          <Column>
            Sort by tags:
          </Column>
          <Column>
            <div
              style={
                {
                  "marginLeft": "10px",
                }
              }
            >
              <select
                className="dropdownmenu"
                id="dropdown"
                onChange={[Function]}
                value="popular"
              >
                <option
                  value="1"
                >
                  Solved
                </option>
                <option
                  value="2"
                >
                  JavaScript
                </option>
                <option
                  value="3"
                >
                  Rust
                </option>
                <option
                  value="4"
                >
                  Linux
                </option>
              </select>
            </div>
          </Column>
          <Column
            right={true}
          >
            <div
              style={
                {
                  "marginRight": "5px",
                }
              }
            >
              Posts with this tag: 
              0
            </div>
          </Column>
        </div>
      </Row>
      <Card
        backgroundColor="rgb(70,70,70)"
        marginBottom={-3}
        title=""
        width="100%"
      >
        <Row />
      </Card>
    </Card>
  </div>
</div>
`;

exports[`Component tests All Posts 1`] = `
<div
  style={
    {
      "alignItems": "stretch",
      "backgroundImage": "linear-gradient(180deg, rgb(110, 160, 175), rgb(15, 40, 60))",
      "display": "flex",
      "flexDirection": "column",
    }
  }
>
  <div
    style={
      {
        "margin": "2%",
      }
    }
  >
    <Card
      backgroundColor="rgb(90,90,90)"
      title=""
      width="100%"
    >
      <Row
        marginBottom={5}
      >
        <div
          style={
            {
              "color": "white",
              "fontSize": "20px",
              "fontWeight": "bold",
              "marginLeft": "5px",
            }
          }
        >
          <Column>
            Sort by:
          </Column>
          <Column>
            <div
              style={
                {
                  "marginLeft": "10px",
                }
              }
            >
              <select
                className="dropdownmenu"
                id="dropdown"
                onChange={[Function]}
                value="popular"
              >
                <option
                  value="popular"
                >
                  Popular
                </option>
                <option
                  value="newest"
                >
                  Newest
                </option>
                <option
                  value="unanswered"
                >
                  Unanswered
                </option>
              </select>
            </div>
          </Column>
          <Column
            right={true}
          >
            <ButtonLight
              onClick={[Function]}
            >
              Sort by tags
            </ButtonLight>
          </Column>
        </div>
      </Row>
      <Card
        backgroundColor="rgb(70,70,70)"
        marginBottom={-3}
        title=""
        width="100%"
      >
        <Row />
      </Card>
    </Card>
  </div>
</div>
`;

exports[`widget tests alert 1`] = `<div />`;
